tools/autograd/load_derivatives.py

```diff
@@ -144,15 +144,25 @@ def set_up_derivatives(defn_name, defn, declaration):
 
         # Set up the derivative information
         derivatives = []
+        args_with_no_gradients = []
```

Comment:
this name is still strange because there are 3 possibilities:
1) an arg has an entry with a specified gradient
2) an arg has an entry specified as not differentiable
3) an arg has no entry

I would expect `args_with_no_gradients` applies to 3), but it actually applies to 2) and this isn't explained.

Reply:
Fair point, let's go for 'not_differentiable_args'.